src/encoder.rs

```diff
@@ -3492,22 +3492,33 @@ pub fn encode_frame<T: Pixel>(
   }
   let tile_group = encode_tile_group(fi, fs, inter_cfg);
 
-  write_obus(&mut packet, fi, fs, inter_cfg).unwrap();
+  if fi.frame_type == FrameType::KEY {
+    write_key_frame_obus(&mut packet, fi, obu_extension).unwrap();
+  }
```

Comment:
Would like another line break after this to separate steps and make reading this easier.

Reply:
Done